xlsx/src/import/worksheets/worksheet_streaming.rs

```diff
@@ -0,0 +1,1034 @@
+        //         </colorScale>
+        //     </cfRule>
+        // </conditionalFormatting>
+        // pageSetup
+        // <pageSetup orientation="portrait" r:id="rId1"/>
+
+        let mut views = HashMap::new();
+        views.insert(
+            0,
+            WorksheetView {
+                row: sheet_view.selected_row,
+                column: sheet_view.selected_column,
+                range: sheet_view.range,
+                top_row: 1,
+                left_column: 1,
+            },
+        );
+
+        Ok((
+            Worksheet {
+                dimension,
+                cols: self.cols,
+                rows: self.rows,
+                name: self.settings.name,
+                sheet_data: self.sheet_data,
+                shared_formulas: self.shared_formulas,
+                sheet_id: self.settings.id,
+                state: self.settings.state,
+                color,
+                merge_cells: self.merge_cells,
+                comments: self.settings.comments,
+                frozen_rows: sheet_view.frozen_rows,
+                frozen_columns: sheet_view.frozen_columns,
+                show_grid_lines: sheet_view.show_grid_lines,
+                views,
+            },
+            sheet_view.is_selected,
+        ))
+    }
+}
+
+fn get_required_attribute_streaming<'a>(
+    tag: &'a BytesStart,
+    attr_name: &str,
+) -> Result<Cow<'a, str>, XlsxError> {
+    tag.try_get_attribute(attr_name)
+        .map_err(|e| {
+            XlsxError::Xml(format!(
+                "Unable to parse attribute: \"{:?}\": {:?}",
+                attr_name,
+                e.to_string()
+            ))
+        })?
+        .ok_or_else(|| {
+            XlsxError::Xml(format!(
+                "Missing required \"{:?}\" XML attribute",
+                attr_name
+            ))
+        })?
+        .unescape_value()
+        .map_err(|e| {
+            XlsxError::Xml(format!(
+                "Unable to decode and unescape attribute: \"{:?}\": {:?}",
+                attr_name,
+                e.to_string()
+            ))
+        })
+}
+
+fn get_optional_attribute_streaming<'a>(
+    tag: &'a BytesStart,
+    attr_name: &str,
+) -> Result<Option<Cow<'a, str>>, XlsxError> {
+    tag.try_get_attribute(attr_name)
+        .map_err(|e| {
+            XlsxError::Xml(format!(
+                "Unable to parse attribute: \"{:?}\": {:?}",
+                attr_name,
+                e.to_string()
+            ))
+        })?
+        .map(|a| a.unescape_value())
+        .transpose()
+        .map_err(|e| {
+            XlsxError::Xml(format!(
+                "Unable to decode and unescape attribute: \"{:?}\": {:?}",
+                attr_name,
+                e.to_string()
+            ))
+        })
+}
+
+fn get_number_streaming(tag: &BytesStart, attr_name: &str) -> i32 {
+    get_optional_attribute_streaming(tag, attr_name)
+        .ok()
+        .unwrap_or(Some("0".into()))
```

Comment:
Why do we need to specify the default three times? I'm new to Rust.

Reply:
We don't, I just don't always remember my Rust combinators.